ext-src/packages/composer/ComposerUtils.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { exec } from "../../utils/exec";
+import { PackageDependenciesHelper } from "../PackageDependenciesHelper";
+import { ComposerPackage } from './ComposerPackage';
+
+const COMPOSER_SHOW_COMMAND = "composer show";
+
+export class ComposerUtils {
+  public async getDependencyArray(): Promise<Array<ComposerPackage>> {
+    console.debug(`running '${COMPOSER_SHOW_COMMAND}'`);
+    try {
+      let { stdout, stderr } = await exec(COMPOSER_SHOW_COMMAND, {
```

Comment:
REALLY late feedback, but if you want to avoid having to run a tool command you could:

open composer.lock
parse the json, it's an array of package with a root packages key.

Should be pretty slick, in that regard, that's what I've been doing in newer tooling (`bach`, `sage`). Also will make the parsing super easy.

Reply:
I had a look: yes, parsing `composer.lock` is easy
but it's not so easy to filter out "require-dev" dependencies